src/Interpreters/ReplaceQueryParameterVisitor.cpp

```diff
@@ -84,16 +94,26 @@ void ReplaceQueryParameterVisitor::visitQueryParameter(ASTPtr & ast)
     IColumn & temp_column = *temp_column_ptr;
     ReadBufferFromString read_buffer{value};
     FormatSettings format_settings;
-    if (ast_param.name == "_request_body")
-        data_type->getDefaultSerialization()->deserializeWholeText(temp_column, read_buffer, format_settings);
-    else
-        data_type->getDefaultSerialization()->deserializeTextEscaped(temp_column, read_buffer, format_settings);
 
-    if (!read_buffer.eof())
-        throw Exception(ErrorCodes::BAD_QUERY_PARAMETER,
-            "Value {} cannot be parsed as {} for query parameter '{}'"
-            " because it isn't parsed completely: only {} of {} bytes was parsed: {}",
-            value, type_name, ast_param.name, read_buffer.count(), value.size(), value.substr(0, read_buffer.count()));
+    try {
+        if (ast_param.name == "_request_body")
+            data_type->getDefaultSerialization()->deserializeWholeText(temp_column, read_buffer, format_settings);
+        else
+            data_type->getDefaultSerialization()->deserializeTextEscaped(temp_column, read_buffer, format_settings);
+
+        if (!read_buffer.eof())
+            throw Exception(ErrorCodes::BAD_QUERY_PARAMETER,
+                "Value {} cannot be parsed as {} for query parameter '{}'"
+                " because it isn't parsed completely: only {} of {} bytes was parsed: {}",
+                value, type_name, ast_param.name, read_buffer.count(), value.size(), value.substr(0, read_buffer.count()));
+    }
+    catch (const Exception & e) {
```

Comment:
Perhaps (didn't check closer) function try blocks may help: https://stackoverflow.com/questions/5612486/when-is-a-function-try-block-useful